src/view/navigation/navigator.ts

```diff
@@ -12,74 +12,101 @@ export type PageType = 'menu' | 'back' | 'empty';
 
 /** Information for each route in app */
 export interface RouteInfo {
-  name: Route;
+  /** Route for the navigator */
+  readonly route: Route;
+  /** The exposed name for the route */
+  readonly name: string;
+  /** The component that should be rendered */
+  readonly component: React.ComponentClass | React.StatelessComponent;
+  /** Specify whether the header/drawer are shown */
+  readonly type: PageType;
+  /** Info provided if page should have search bar */
+  readonly searchInfo?: SearchInfo;
+  /** Information provided if route should appear in drawer */
+  readonly drawerInfo?: DrawerInfo;
+}
+
+/** Information for route containing drawer entry */
+export interface DrawerRouteInfo extends RouteInfo {
+  readonly drawerInfo: DrawerInfo;
+}
+
+/** Information for route with search bar */
+export interface SearchRouteInfo extends RouteInfo {
+  readonly searchInfo: SearchInfo;
+}
+
+interface DrawerInfo {
   icon: string;
-  component: React.ComponentClass | React.StatelessComponent;
-  inDrawer: boolean;
-  type: PageType;
-  search: boolean;
-  searchPlaceholder?: string;
+}
+
+interface SearchInfo {
+  placeholder: string;
 }
 
 /** Named routes in the application */
 export enum Route {
   HOME = 'Home',
   LOGIN = 'Login',
   EXPORTS = 'Exports',
-  FARMER = 'FarmerInfo',
-  SEARCH_FARMER = 'Farmers',
+  FARMER = 'Farmer',
+  SEARCH_FARMER = 'SearchFarmers',
```

Comment:
thanks for fixing this! 👍 
might need to make search_farmer and searchfarmerS consistent though?

Reply:
you mean like `search_farmers` and `farmer`?